packages/as-types/assembly/serialization/numbers.ts

```diff
@@ -20,17 +20,35 @@ export function byteToU8(arr: StaticArray<u8>): u8 {
   return arr[0];
 }
 
+/**
+ * Convert a number to StaticArray<u8>
+ *
+ *  @param val - the number to convert
+ */
+export function toBytes<T>(val: T): StaticArray<u8> {
+  assert(isInteger<T>(), 'input must be a integer');
```

Comment:
better to cause compilation error
```suggestion
  if (!isInteger<T>()) {
    ERROR('input must be a integer');
  }
```

Reply:
nice !